activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -156,6 +156,10 @@ def isolation
         @isolation_level
       end
 
+      def isolation=(isolation) # :nodoc:
+        @isolation_level = isolation
```

Comment:
We should probably assert that this transaction isn't joinable, because this API used with real transactions would be pretty bad.

From the DB point of view this is a NOOP, setting it does nothing except change the return value of `#isolation`, however, in case of a query retry, we might reconnect and reopen transactions, which would cause the real isolation level to change. That would be super surprising.

Reply:
Addressed this above.